### ISSUE TITLE
Create ARI summary dataframe prior to summary plot

### ISSUE THREAD

**Issue by allyhawkins** [AlexsLemonade/scpca-downstream-analyses#102]
**Please provide some background on the proposed additions or changes.**

Ultimately we would like to provide informative summary plots to show clustering stability across the methods of clustering and the parameter values being tested. To measure cluster stability, we have performed bootstrapping of the cells and then re clustered the cells and compared the re-clustering of the cells to the original cluster assignments. To measure the comparisons we are calculating the ARI and returning a single ARI for each iteration of bootstrapping. After which we need to compile this into informative plots. Before we can create the summary plot, we will need to create a single dataframe that combines the ARI's across clustering methods and summarizes them so we have the median and MAD for each clustering method and param value tested. 

**What are the changes that you are proposing?**
 

Based on the discussion in #95, we will want to create a summary plot similar to the plots we have that summarize purity and silhouette width across clustering methods and parameter values tested. We would eventually like to produce one line plot that combines all clustering types together and looks at the median ARI across bootstrapping iterations and plots those median values across parameter values that were tested. First we will need to create the dataframe used for plotting which should combine multiple dataframes produced from #99 using different clustering methods. 

**Please describe the proposed solution.**

We should have a separate plotting function that takes as input a list of dataframes that are created in #99, one for each type of clustering method that is tested. The dataframes will need to be combined and then the summary statisitcs need to be calculated to create a dataframe that contains the median values and the MAD values (as calculated previously in [`summarize_clustering_stats()`](https://github.com/AlexsLemonade/scpca-downstream-analyses/blob/9e08dcd48fe8b409d794fdf28c7a84d026d1dcbc/utils/clustering-functions.R#L256). Here we should just focus on inputting the dataframes and creating the combined dataframe with the correct values to be plotting. We will create a separate issue for actually creating the summary plot.
